src/frontend/js/pages/TeacherDashboardContractsLoader/TeacherDashboardContracts/index.tsx

```diff
@@ -1,144 +1,184 @@
+  const contractToSignCount = halfSignedContractsQuery.meta?.pagination?.count ?? 0;
+  const hasContractToSign = contractToSignCount > 0;
 
   const {
     items: contracts,
     meta,
-    states: { fetching, isFetched },
-  } = useContracts(contractFilters);
-
-  const [sortModel, setSortModel] = useState<SortModel>([
-    {
-      field: 'learnerName',
-      sort: 'desc',
-    },
-    {
-      field: 'productTitle',
-      sort: 'desc',
-    },
+    states: { fetching, isFetched, error },
+  } = useContracts(
     {
-      field: 'signDate',
-      sort: 'desc',
+      ...filters,
+      course_id: courseId,
+      product_id: productId,
+      page: pagination.page,
+      page_size: PER_PAGE.teacherContractList,
     },
-  ]);
+    { enabled: !!filters.organization_id },
+  );
+
+  const rows = useMemo(() => {
+    return contracts.map((contract) => ({
+      id: contract.id,
+      learnerName: contract.order.owner_name,
+      productTitle: contract.order.product_title,
+      state: ContractHelper.getHumanReadableState(contract, ContractStatePoV.ORGANIZATION, intl),
+    }));
+  }, [contracts]);
+
+  const handleFiltersChange = (newFilters: Partial<ContractFilters>) => {
+    // Reset pagination
+    pagination.setPage(1);
+    setFilters((prevFilters) => ({ ...prevFilters, ...newFilters }));
+  };
 
-  const pagination = usePagination({
-    defaultPage: page ? parseInt(page, 10) : 1,
-    pageSize: PER_PAGE.teacherContractList,
-  });
   useEffect(() => {
-    if (isFetched) {
+    if (isFetched && meta?.pagination?.count) {
       pagination.setPagesCount(Math.ceil(meta!.pagination!.count / PER_PAGE.teacherContractList));
     }
-  }, [contracts, isFetched]);
-
-  const onPageChange = (newPage: number) => {
-    setContractFilters((currentFilters: ContractFilters) => ({ ...currentFilters, page: newPage }));
-  };
-  useEffect(() => onPageChange(pagination.page), [pagination.page]);
+  }, [meta, isFetched]);
 
-  const rows = useMemo(() => {
-    return contracts.map(({ id, student_signed_on: studentSignedOn, order }) => ({
-      id,
-      learnerName: order.owner_name,
-      productTitle: order.product_title,
-      signDate: studentSignedOn ? intl.formatDate(studentSignedOn) : studentSignedOn,
-    }));
-  }, [contracts]);
+  if (error) {
+    return <Banner message={error} type={BannerType.ERROR} rounded />;
+  }
 
-  return contracts ? (
-    <div className="teacher-contract-page" data-testid={isFetched ? 'contracts-loaded' : undefined}>
+  return (
+    <div className="teacher-contract-page">
+      <div className="dashboard__page__actions">
+        <div className="dashboard__page__actions-row dashboard__page__actions-row--space-between">
+          <div>
+            {filters.organization_id && canSignContract && (
+              <>
+                {hasContractToSign && (
+                  <Button
+                    size="small"
+                    color="secondary"
```

Comment:
Semantic: it seems to be the main action of this page, so it may need to be primary? WDYT?

Reply:
Yup indeed !